code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -0,0 +1,119 @@
+/obj/machinery/smoke_machine
+	name = "Smoke Machine"
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "smoke0"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/smoke_machine
+	var/efficiency = 10
+	var/on = FALSE
+	var/cooldown = 0
+	var/screen = "home"
+	var/useramount = 30 // Last used amount
+	var/volume = 1000
+	var/setting = 3
+	var/list/possible_settings = list(3,6,9,12,15)
+
+/datum/effect_system/smoke_spread/chem/smoke_machine/set_up(datum/reagents/carry = null, setting = 3, efficiency = 10, loc)
+	amount = setting
+	carry.copy_to(chemholder, 20)
+	carry.remove_any(setting * 16 / efficiency)
+	location = loc
+
+/obj/machinery/smoke_machine/Initialize()
+	. = ..()
+	create_reagents(volume)
+
+/obj/machinery/smoke_machine/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
```

Comment:
I don't see a START_PROCESSING anywhere?

Reply:
The machinery code does it somewhere.